plugin/evm/limit_order.go

```diff
@@ -209,7 +210,30 @@ func (lop *limitOrderProcesser) listenAndStoreLimitOrderTransactions() {
 				executeFuncAndRecoverPanic(func() {
 					lop.mu.Lock()
 					defer lop.mu.Unlock()
+
+					if len(logs) == 0 {
+						return
+					}
+
+					blockNumber := logs[0].BlockNumber
+					block := lop.blockChain.GetBlockByHash(logs[0].BlockHash)
+
+					// If n is the block at which snapshot should be saved(n is multiple of [snapshotInterval]), save the snapshot
+					// when logs of block number >= n + 1 are received before applying them in memory db
+
+					blockNumberFloor := ((blockNumber - 1) / snapshotInterval) * snapshotInterval
+					if blockNumberFloor > lop.snapshotSavedBlockNumber {
+						lop.memoryDb.Accept(blockNumberFloor, block.Timestamp())
+						err := lop.saveMemoryDBSnapshot(big.NewInt(int64(blockNumberFloor)))
+						if err != nil {
+							log.Error("Error in saving memory DB snapshot", "err", err)
+						}
+						lop.snapshotSavedBlockNumber = blockNumberFloor
+					}
+
 					lop.contractEventProcessor.ProcessAcceptedEvents(logs, false)
+					lop.memoryDb.Accept(blockNumber, block.Timestamp())
+
 					if lop.tradingAPIEnabled {
 						go lop.contractEventProcessor.PushtoTraderFeed(logs, orderbook.ConfirmationLevelAccepted)
 						go lop.contractEventProcessor.PushToMarketFeed(logs, orderbook.ConfirmationLevelAccepted)
```

Comment:
should we send these events first thing?

Reply:
yep, it's best to send them first. Fixed now